pyvista/plotting/charts.py

```diff
@@ -0,0 +1,3375 @@
+        >>> chart.show()
+
+        """
+        return self.GetTicksVisible()
+
+    @ticks_visible.setter
+    def ticks_visible(self, val):
+        self.SetTicksVisible(bool(val))
+
+    def _update_ticks(self):
+        locs = None if self._tick_locs.GetNumberOfValues() == 0 else self._tick_locs
+        labels = None if self._tick_labels.GetNumberOfValues() == 0 else self._tick_labels
+        self.SetCustomTickPositions(locs, labels)
+
+
+class _CustomContextItem(_vtk.vtkPythonItem):
+
+    class ItemWrapper(object):
+
+        def Initialize(self, item):
+            # item is the _CustomContextItem subclass instance
+            return True
+
+        def Paint(self, item, painter):
+            # item is the _CustomContextItem subclass instance
+            return item.paint(painter)
+
+    def __init__(self):
+        super().__init__()
+        self.SetPythonObject(_CustomContextItem.ItemWrapper())  # This will also call ItemWrapper.Initialize
+
+    def paint(self, painter):
+        return True
+
+
+class _ChartBackground(_CustomContextItem):
+    """Utility class for chart backgrounds (until native VTK support is available)."""
+
+    def __init__(self, chart):
+        super().__init__()
+        self._chart = chart  # Chart to draw the background for
+        # Default background is translucent with black border line
+        self.BorderPen = Pen(color=(0, 0, 0))
+        self.BackgroundBrush = Brush(color=(0, 0, 0, 0))
+
+    def paint(self, painter):
+        if self._chart.visible:
+            painter.ApplyPen(self.BorderPen)
+            painter.ApplyBrush(self.BackgroundBrush)
+            l, b, w, h = self._chart._geometry
+            painter.DrawRect(l, b, w, h)
+            # TODO: following 'patch' is necessary until vtkPlotPie is fixed. Otherwise Pie plots will use the same
+            #  opacity as the chart's background when their legend is hidden. As the default background is transparent,
+            #  this will cause Pie charts to completely disappear.
+            painter.GetBrush().SetOpacity(255)
+        return True
+
+
+class _Chart(object):
+    """Common pythonic interface for vtkChart, vtkChartBox, vtkChartPie and ChartMPL instances."""
+
+    def __init__(self):
+        super().__init__()
+        self._background = _ChartBackground(self)
+        self._x_axis = Axis()  # Not actually used for now (see note in Chart2D), but still present for the
+        self._y_axis = Axis()  # Charts.toggle_interaction code
+        self._z_axis = Axis()
+
+    @property
+    def _scene(self):
+        """Get a reference to the vtkScene in which this chart is drawn."""
+        return self.GetScene()
+
+    @property
+    def _renderer(self):
+        """Get a reference to the vtkRenderer in which this chart is drawn."""
+        return self._scene.GetRenderer()
+
+    def _render_event(self, *args, **kwargs):
+        """This method will be called before every render."""
```

Comment:
The problem here is that docstring excerpt lines should be phrased as "Do the thing". Instead you have something very different. You should say someting about the functionality of the function on this line, and then mention it being called each time on a separate row.